cmd/service-controller/main.go

```diff
@@ -61,32 +62,43 @@ func main() {
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := SetupSignalHandler()
 
-	// todo, get context from env?
-	cli, err := client.NewClient(namespace, "", "")
-	if err != nil {
-		log.Fatal("Error getting van client", err.Error())
-	}
-
 	tlsConfig, err := certs.GetTlsConfig(true, types.ControllerConfigPath+"tls.crt", types.ControllerConfigPath+"tls.key", types.ControllerConfigPath+"ca.crt")
 	if err != nil {
 		log.Fatal("Error getting tls config", err.Error())
 	}
 
-	event.StartDefaultEventStore(stopCh)
-
-	controller, err := NewController(cli, origin, tlsConfig, disableServiceSync == "true")
-	if err != nil {
-		log.Fatal("Error getting new controller", err.Error())
-	}
-
-	log.Println("Waiting for Skupper router component to start")
-	_, err = kube.WaitDeploymentReady(types.TransportDeploymentName, namespace, cli.KubeClient, time.Second*180, time.Second*5)
-	if err != nil {
-		log.Fatal("Error waiting for transport deployment to be ready: ", err.Error())
-	}
-
-	// start the controller workers
-	if err = controller.Run(stopCh); err != nil {
-		log.Fatal("Error running controller: ", err.Error())
+	platform := config.GetPlatform()
```

Comment:
I wonder if it might be simpler here to have a separate main.go for the podman version, and therefore a separate image?

Reply:
That is a good question.
I would say yes, it would be simpler and probably safer, especially for the kubernetes controller.
The only painpoint, or maybe just something to consider, is that it introduces an extra image to be maintained.

Suppose we split it out as a separate process/image, what would be your preference to place it?
* cmd/service-controller-podman, or,
* cmd/podman/service-controller ?   <-- this would be my choice

Considering we change it to be a separate process/image, I believe we should do the same to
the flow-collector process/image. What do you think @grs @ajssmith @nluaces ?

Reply:
@grs could you please review this alternative PR, which introduces a new image ( #1120 )?